main.go

```diff
@@ -83,8 +83,8 @@ func main() {
 		fmt.Sprintf(
 			"The type of client cache persistence model that should be employed."+
 				"choices=%v", []string{persistenceModelDirectUnencrypted, persistenceModelDirectEncrypted, persistenceModelNone}))
-	flag.IntVar(&vdsOptions.MaxConcurrentReconciles, "max-concurrent-reconciles-vds", 100,
-		"Maximum number of concurrent reconciles for the VaultDynamicSecrets controller.")
+	flag.IntVar(&controllerOptions.MaxConcurrentReconciles, "max-concurrent-reconciles-global", 100,
```

Comment:
I wonder if we could rename the option to `max-concurrent-reconciles`, since global seems to imply that we support per controller reconciles. Also, I wonder if we might be breaking anyone who has set `max-concurrent-reconciles-vds`, should we maybe leave that around as a deprecated option. We could assign its value to the option?

Reply:
Sounds good about renaming to `max-concurrent-reconciles` and applying to all controllers.

Keeping the old flag is maybe a bit hacky since we don't have a way of knowing if the flag was even passed due to the way the `flags.Int()` function takes a default value. I could set it to `0` by default and ignore it unless it's non-zero I suppose.
At any rate, this would only presumably be a problem for someone who's using Kustomize and had set the flag, and for those who are using the helm chart but pinning old versions of the operator binary. I think that second one is a bit non-standard and questionably supportable?

I guess I wonder if it's a trade-off of documenting it as a breaking change for Kustomize users, and/or making the code a little sloppy? Thoughts?

Reply:
Yeah, keeping it around is not the best, but I think it is possible to set it from `controller.manager.extraArgs` here: https://github.com/hashicorp/vault-secrets-operator/blob/5845d8abe3d96fe9b9e3b6c27d1bce6aef78e020/chart/values.yaml#L330 - we can document this as a breaking change, or keep it as an alias to the new option.

Reply:
Okay - moved everything to `max-concurrent-reconciles`, preserving `max-concurrent-reconciles-vds` which will override `max-concurrent-reconciles` if supplied.